### DISCUSSION TITLE
Throw across multiple monitors

### DISCUSSION THREAD

**Discussion by certainlyakey** [rxhanson/RectanglePro-Community#221] [Q&A]
I'm using this amazing tool, but one thing puzzles me: when I use either Window Throw or Long Throw, neither option allows me to move windows across monitors. 

I have a dual monitor setup with 2 monitors stacked horizontally, and it feels so right to throw a window to another monitor by moving the mouse there. But it seems impossible: the gray window overlay only appears on the monitor where the window being moved is on. 

Am i missing something and it's actually possible with some config tweak? Or is it a technical limitation?

**Comment by rxhanson** [comment]
Thanks for the feedback. If I understand correctly, your expectation is that when you move your cursor to the other display, you are expecting an overlay and result to appear on the other display.

I realize this seems odd, but I just haven't gotten to it yet. There's a few different ways to go about it, and I never decided which one was right. Some options:
* Change the initial cursor position (the red dot) to the center of the new display. This seems like weird behavior, especially in scenarios like moving the cursor back to the previous display and having to reset the initial cursor position.
* Give the user a dropdown and let them pick which window size/position goes on the new display. Less flexibility, but less ambiguity & more clarity.

I think the 2nd one is my choice.

What is your expected behavior on the other display once your cursor gets there?

**Reply by rxhanson** [reply]
(In the meantime, you could select next display or previous display for the long throw)

**Comment by certainlyakey** [comment]
Maybe i have a different mental modal of how window throw works. But I was thinking actually about the same approach as now: 

1. you press the modifiers
2. start moving your mouse around
3. If it enters the corner/top/bottom areas on the current screen, related action commences
4. if it enters the corner/top/bottom areas on a different screen, related action commences again, according to the configuration. 

Eg if i have monitors A and B and have configured my mouse to trigger left half resizing on moving it to top-left, the trigger should work regardless whether it's top-left of monitor A or monitor B. 

I hope it makes sense...

**Reply by rxhanson** [reply]
Window throw works a little differently in that all of the sizes/positions are relative to the initial cursor position, rather than absolute positioning on the screen. This has advantages in that there is typically less cursor movement involved, resulting in faster actions, and overlapping areas can be more easily discerned. There are disadvantages, one being that a lot of users think of it in terms of absolute screen positioning of the cursor, which can be confusing.

Another thought I had was to have a separate configuration for absolute screen positioning, and use that when the user reaches another display with the cursor, but that's not straightforward either without some form of additional communication to the user, maybe something like lines on the screen. Really this boils down to a different feature that I've only partially implemented at this time: a Rectangle Pro style Fancy Zones implementation. 

So, going back to the original problem with Window Throw - the expectation that when the cursor reaches the other display that there will be some form of window size/position that is on that display. I think I prefer just a dropdown for what should happen on the other display. That way you can, say, set it to maximize on the other display, and quickly reposition with another window throw if needed.

**Reply by certainlyakey** [reply]
OK, thanks for the explanation. I do hope this feature makes it in whatever form you see is best adapted to minimise the effort of the RP users. 

I usually don't use multiple windows in one screen mode (eg using laptop on battery only, unconnected to an external monitor - the laptop screen is simply too small to my taste for that), so window throw is only valuable for multiple monitors configurations. But it is still useful to have it even on one monitor.
